test/unit/QueueConsumer-test.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * QueueConsumer unit test
+ */
+require('require-self-ref')
+
+const test = require('ava')
```

Comment:
I'm liking `ava` so far. I kind of miss the ability to group tests via `describe` or `context`. Not a huge deal though.